tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,379 @@
+import logging
+import os
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+from unittest.case import TestCase
+
+import boto3
+import pytest
+import yaml
+from botocore.exceptions import ClientError
+from botocore.config import Config
+from tests_integ.helpers.deployer.deployer import Deployer
+from tests_integ.helpers.helpers import transform_template, verify_stack_resources, generate_suffix, create_bucket
+from tests_integ.helpers.file_resources import FILE_TO_S3_URI_MAP, CODE_KEY_TO_FILE_MAP
+
+LOG = logging.getLogger(__name__)
+STACK_NAME_PREFIX = "sam-integ-stack-"
+S3_BUCKET_PREFIX = "sam-integ-bucket-"
+
+
+class BaseTest(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.tests_integ_dir = Path(__file__).resolve().parents[1]
+        cls.resources_dir = Path(cls.tests_integ_dir, "resources")
+        cls.template_dir = Path(cls.resources_dir, "templates", "single")
+        cls.output_dir = Path(cls.tests_integ_dir, "tmp")
+        cls.expected_dir = Path(cls.resources_dir, "expected", "single")
+        cls.code_dir = Path(cls.resources_dir, "code")
+        cls.s3_bucket_name = S3_BUCKET_PREFIX + generate_suffix()
+        cls.session = boto3.session.Session()
+        cls.my_region = cls.session.region_name
+        cls.s3_client = boto3.client("s3")
```

Comment:
Should we be creating clients based on the session instead? I think this is typically the better practice, though it probably doesn't impact anything here. Just caught my eye is all.

Reply:
Yep this could be better indeed, to be sure we are using the same session throughout

Reply:
By the boto3 documentation: "Boto3 acts as a proxy to the default session. This is created automatically when you create a low-level client." So all the clients are created by using the default session and during the tests we don't use any customized session.